### DISCUSSION TITLE
How to evaluate the answer [Q&A Bot] ?

### DISCUSSION THREAD

**Discussion by alexmao13** [FlowiseAI/Flowise#701] [Q&A]
Hi, I am building a Q&A Chatbot that intended to ask user customer questions and use a chain to evaluate the answer and decide if the next/new question should be asked.  so take history via API can be complicated, I am seeking for away of getting it within the Flowise.

I also need save the conversation history to a database.  The problem for me is I am unable to find a way to extract the conversation history. Would be great if anyone can help ?

**Comment by chungyau97** [comment]
Hi @alexmao13,

It's on our roadmap to identify next/new/end conversation based on user's message; It's on discussion stage.

**Embed chat**, we are currently working storing the chat history. 
**Your own chat system**, you will need to keep track it yourself and send it back to [Flowise API](https://docs.flowiseai.com/how-to-use#api).
